sdk/attestation/azure-security-attestation/test/ut/attestation_test.cpp

```diff
@@ -69,97 +80,132 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
 
-    auto attestationSigners = attestationClient->GetAttestationSigningCertificates();
-    EXPECT_LE(1UL, attestationSigners.Value.Signers.size());
-    for (const auto& signer : attestationSigners.Value.Signers)
+    if (std::get<1>(GetParam()) == AttestationType::OpenEnclave)
     {
-      EXPECT_TRUE(signer.KeyId.HasValue());
-      EXPECT_LE(1UL, signer.CertificateChain.Value().size());
-      for (const auto& cert : signer.CertificateChain.Value())
-      {
-        EXPECT_EQ(0UL, cert.find("-----BEGIN CERTIFICATE-----\r\n"));
-      }
+      auto report = AttestationCollateral::OpenEnclaveReport();
+      auto attestResponse = client->AttestOpenEnclave(report);
+      ValidateAttestResponse(attestResponse);
+    }
+    else if (std::get<1>(GetParam()) == AttestationType::SgxEnclave)
+    {
+      auto quote = AttestationCollateral::SgxQuote();
+      auto attestResponse = client->AttestSgxEnclave(quote);
+      ValidateAttestResponse(attestResponse);
     }
-  }
 
-  TEST_P(AttestationTests, SimpleAttestOpenEnclave)
-  {
-    auto client(CreateClient());
     auto report = AttestationCollateral::OpenEnclaveReport();
-
     auto attestResponse = client->AttestOpenEnclave(report);
   }
 
-  TEST_P(AttestationTests, SimpleAttestSgxEnclave)
-  {
-    auto client(CreateClient());
-    auto sgxQuote = AttestationCollateral::SgxQuote();
-
-    auto attestResponse = client->AttestSgxEnclave(sgxQuote);
-  }
-
-  TEST_P(AttestationTests, AttestOpenEnclaveWithRuntimeData)
+  TEST_P(AttestationTests, AttestWithRuntimeData)
   {
     auto client(CreateClient());
-    auto report = AttestationCollateral::OpenEnclaveReport();
     auto runtimeData = AttestationCollateral::RuntimeData();
 
-    auto attestResponse = client->AttestOpenEnclave(
-        report, {AttestationData{runtimeData, AttestationDataType::Binary}});
+    if (std::get<1>(GetParam()) == AttestationType::OpenEnclave)
+    {
+      auto report = AttestationCollateral::OpenEnclaveReport();
+      AttestationData data{runtimeData, AttestationDataType::Binary};
+      auto attestResponse = client->AttestOpenEnclave(report, {data});
+      ValidateAttestResponse(attestResponse, data);
+    }
+    else if (std::get<1>(GetParam()) == AttestationType::SgxEnclave)
+    {
+      auto quote = AttestationCollateral::SgxQuote();
+      AttestationData data{runtimeData, AttestationDataType::Binary};
+      auto attestResponse = client->AttestSgxEnclave(quote, {data});
+      ValidateAttestResponse(attestResponse, data);
+    }
   }
-
-  TEST_P(AttestationTests, AttestSgxEnclaveWithRuntimeData)
+  TEST_P(AttestationTests, AttestWithRuntimeDataJson)
   {
     auto client(CreateClient());
-    auto sgxQuote = AttestationCollateral::SgxQuote();
     auto runtimeData = AttestationCollateral::RuntimeData();
 
-    auto attestResponse = client->AttestSgxEnclave(
-        sgxQuote, {AttestationData{runtimeData, AttestationDataType::Binary}});
+    if (std::get<1>(GetParam()) == AttestationType::OpenEnclave)
```

Comment:
save `std::get<1>(GetParam())` into a const var and then do the if/else if

Reply:
Why a `const` variable instead of just a local? There is no perf or behavior benefit as far as I can tell.

Reply:
FWIW, there are very real benefits to byref const parameters. But I'm not sure there are similar benefits to const locals, especially when the const local is POD (yes, I know that enumerable strings aren't POD).

Reply:
my bad, what I meant is:

auto const& param = std::get<1>(GetParam());

Then you use `param` instead of keep calling `std::get<1>(GetParam())` multiple times.  It's just a tiny code enhance

Reply:
My question was not about pulling the value out as a local variable - I think that's a great idea. My question was why do you want you declare `auto const& param` instead of `auto param` or `std::string param`. What value does the `const&` bring to the code? Also, using *const&* adds *significant* overhead if the value is a scalar (which is it for many of these cases). 

Oh, and also: declaring this as a reference variable makes some assumptions about the internal implementation of `std::get<>` and I'm not sure those  assumptions are valid.